pull-data.mjs

```diff
@@ -335,8 +367,16 @@ export default async function run({ github, context, dryRun = false }) {
     return;
   }
 
-  const builds = existingBuilds.concat(newBuilds);
-
+  const builds = [...existingBuilds]
+  for (const b of newBuilds) {
+    const index = builds.findIndex((b2) => b2.build_number === b.build_number);
+    if (index !== -1) {
+      b.created_at = builds[index].created_at;
+      builds[index] = b;
+    } else {
+      builds.push(b);
+    }
+  }
```

Comment:
what is this trying to protect against? when would we get a build in both existing and new builds?